Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -701,6 +704,10 @@ void MediaPlayerPrivateGStreamerMSE::updateStates()
                     m_readyState = MediaPlayer::HaveEnoughData;
                     LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
                     m_networkState = m_downloadFinished ? MediaPlayer::Idle : MediaPlayer::Loading;
+                } else if (m_mediaSource) {
```

Comment:
It would be great to refactor this new branch into the original "else" branch below. Both branches do the same except fro the monitorSourceBuffers() and the m_readyState assignment.